autoload/ThemeManager.gd

```diff
@@ -1,102 +1,100 @@
 extends Node
 
 const THEME_ROOT := "res://ui/theme/"
-const THEME_FONTS_ROOT := "res://ui/theme/fonts/"
+const THEME_TYPES: Array[String] = ["Label", "Button", "LineEdit", "TextEdit", "RichTextLabel", "CheckBox", "ItemList"]
 
 const COLOR_TEXT_DEFAULT := Color(0.960784, 0.980392, 0.980392)
 const COLOR_TEXT_LOWER_CONTRAST := Color(0.736288, 0.728113, 0.839844)
 
-onready var _theme = preload("res://ui/theme/gdscript_app_theme.tres")
-
-var _font_defaults := {}
+@onready var _theme: Theme = preload("res://ui/theme/gdscript_app_theme.tres")
 
+var _theme_defaults: Dictionary = {}
 
 func _ready() -> void:
-	_cache_font_defaults()
-	
-	var current_profile := UserProfiles.get_profile()
-	scale_all_font_sizes(current_profile.font_size_scale, false)
-	set_lower_contrast(current_profile.lower_contrast, false)
-	set_dyslexia_font(current_profile.dyslexia_font, false)
-
-
-func _cache_font_defaults() -> void:
-	_font_defaults.clear()
+	_cache_theme_defaults()
```

Comment:
Here I recommend inlining the function's code and not calling a separate function. You can include a comment about why you use this code: to store the default size of things.

Splitting small pieces of code like that to separate functions, on the one hand, labels the functions, but when they're small single use pieces like this, it also creates indirection. In a small script like that, it's fine. It's just at scale, it makes the code harder to trace.